app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt

```diff
@@ -115,6 +126,45 @@ class FinishedFlightsViewModel(val repository: Repository, val userId: String) :
             }
       }
 
+  private fun searchLocation(query: String) =
+      viewModelScope.launch(Dispatchers.IO) {
+        val client = OkHttpClient()
+        val url = "https://nominatim.openstreetmap.org/search?q=$query&format=json&limit=4"
+        val request = Request.Builder().url(url).build()
+        _searchResponse.value =
+            try {
+              val response = client.newCall(request).execute()
+              val responseBody = response.body()?.string()
+              responseBody
+            } catch (e: Exception) {
+              e.printStackTrace()
+              "error"
+            }
+      }
+
+  fun getSearchLocation(query: String, time: Int) =
+      viewModelScope.launch {
+        searchLocation(query).join()
```

Comment:
You might want to make the `searchLocation` a suspend function, as you are already starting a coroutine in `getSearchLocation` and you immediately join it anyway

Reply:
Done